lib/python-indent.js

```diff
@@ -294,4 +294,43 @@ export default class PythonIndent {
         // Set the indent
         this.editor.setIndentationForBufferRow(row, indent);
     }
+
+    reindent() {
+        this.editor = atom.workspace.getActiveTextEditor();
+
+        if (!this.editor ||
+            this.editor.getGrammar().scopeName.substring(0, 13) !== "source.python" ||
+            !this.editor.getSoftTabs()) {
+            return;
+        }
+
+        let selectedText = this.editor.getSelectedText();
+
+        // if last char is newline, remove it.
+        if (selectedText[selectedText.length - 1] === "\n") {
+            selectedText = selectedText.slice(0, -1);
+        }
+
+        const selectedTextLines = selectedText.split("\n");
+
+        if (!selectedTextLines.length) {
+            return;
+        }
+
+        // Delete the currently selected text. The documentation says:
+        //    For *each* selection, if the selection is empty,
+        //    delete the character following the cursor.
+        //    Otherwise delete the selected text.
+        // but the documentation for editor.getSelectedText() says:
+        //    Get the selected text of the *most recently added* selection.
+        // (emphasis mine in both quotes) so that's a little concerning...
+        this.editor.delete();
+
+        for (let row = 0; row < selectedTextLines.length; row += 1) {
+            const line = selectedTextLines[row];
+            this.editor.insertText(line.trim());
+            const editorElement = atom.views.getView(atom.workspace.getActiveTextEditor());
```

Comment:
I think you can use `this.editor` here instead of `atom.workspace.getActiveTextEditor`.